docs/src/restarts.md

```diff
@@ -0,0 +1,101 @@
+## Restarting Simulations
+
+ClimaLand provides functionality to save and load simulation checkpoints,
+allowing you to restart simulations from a previous state. This is particularly
+useful for long-running simulations or if you want to experiment with different
+configurations starting from a specific point in the simulation.
+
+
+### Saving Checkpoints
+
+To save a simulation checkpoint, you can use the `ClimaLand.save_checkpoint`
+function. This function takes the current state Y, the simulation time t, and
+the output directory as arguments. Optionally you can provide the ClimaLand
+model object model. This will store the hash of the model in the checkpoint
+file. You can use this information to ensure that you are restarting the
+simulation with the same model that was used to generate the checkpoint.
+
+```julia
+ClimaLand.save_checkpoint(Y, t, output_dir; model)
+```
+
+Most typically, this function is not called directly. Instead, it is called as a
+callback.
+
+In ClimaLand, you can automate the process of saving checkpoints using the
+`CheckpointCallback`. This callback allows you to specify the frequency at which
+checkpoints are saved and handles the saving process during the simulation.
+
+To use the CheckpointCallback, you need to create an instance of it and pass
+it to the solve function along with your other callbacks.
+
+Example:
+
+```julia
+
+# ... your ClimaLand simulation setup ...
+
+# Create a CheckpointCallback to save checkpoints every 6 hours
+checkpoint_cb = CheckpointCallback(Dates.Hour(6), output_dir, start_date, t_start; model)
```

Comment:
Do we need to give any guidance re: the timestep dividing the checkpoint interval with no remainder?

Reply:
Yes, I will add those checks in my next PR

Reply:
I added it here.